tests/handler_tests/test_weaviate_handler.py

```diff
@@ -0,0 +1,496 @@
+# check if weaviate is installed
+import importlib
+from unittest.mock import patch
+
+import pandas as pd
+import pytest
+from mindsdb_sql import parse_sql
+
+from ..unit.executor_test_base import BaseExecutorTest
+
+try:
+    importlib.import_module("weaviate")
+    WEAVIATE_INSTALLED = True
+except ImportError:
+    WEAVIATE_INSTALLED = False
+
+
+@pytest.mark.skipif(not WEAVIATE_INSTALLED, reason="weaviate is not installed")
+class TestWeaviateHandler(BaseExecutorTest):
+    def run_sql(self, sql):
+        ret = self.command_executor.execute_command(parse_sql(sql, dialect="mindsdb"))
+
+        assert ret.error_code is None
+        if ret.data is not None:
+            columns = [
+                col.alias if col.alias is not None else col.name for col in ret.columns
+            ]
+            return pd.DataFrame(ret.data, columns=columns)
+
+    def setup_method(self):
+        super().setup_method()
+        # create a weaviate database connection
+        self.run_sql(
+            f"""
+            CREATE DATABASE weaviate_test
+            WITH ENGINE = "weaviate",
+            PARAMETERS = {{
+                "weaviate_url" : "{"https://sample-q4x5xe6o.weaviate.network"}",
+                "weaviate_api_key": "{"QBHA2JJ3PxXF6Rs8cl3a6IsYjp0jDFh0sehv"}"
+            }}
```

Comment:
@abhijitpal1247 I would advise against hard coding api keys in tests.

I would recommend we replace this with connection to locally running weaviate docker for testing

Reply:
resolved in e8944f0fe38ecffba0a915419bd098a57c562e55